sunpy/coordinates/sun.py

```diff
@@ -0,0 +1,394 @@
+"""
+Sun-specific coordinate calculations
+"""
+import warnings
+
+import numpy as np
+
+import astropy.units as u
+from astropy.time import Time
+from astropy.coordinates import Angle, Latitude, Longitude, SkyCoord
+# Versions of Astropy that do not have *MeanEcliptic frames have the same frames
+# with the misleading names *TrueEcliptic
+try:
+    from astropy.coordinates import HeliocentricMeanEcliptic, GeocentricMeanEcliptic
+except ImportError:
+    from astropy.coordinates import HeliocentricTrueEcliptic as HeliocentricMeanEcliptic
+    from astropy.coordinates import GeocentricTrueEcliptic as GeocentricMeanEcliptic
+from astropy import _erfa as erfa
+from astropy.coordinates.builtin_frames.utils import get_jd12
+
+from sunpy.sun import constants
+from sunpy.time import parse_time
+from sunpy.time.time import _variables_for_parse_time_docstring
+from sunpy.util.decorators import add_common_docstring
+from .ephemeris import get_earth, _B0, _L0, _P, _earth_distance, _orientation
+
+__all__ = [
+    "angular_radius", "sky_position", "carrington_rotation_number",
+    "true_longitude", "apparent_longitude", "true_latitude", "apparent_latitude",
+    "mean_obliquity_of_ecliptic", "true_rightascension", "true_declination",
+    "true_obliquity_of_ecliptic", "apparent_rightascension", "apparent_declination",
+    "print_params",
+    "B0", "L0", "P", "earth_distance", "orientation"
+]
+
+
+@add_common_docstring(**_variables_for_parse_time_docstring())
+def angular_radius(t='now'):
+    """
+    Return the angular radius of the Sun as viewed from Earth.
+
+    Parameters
+    ----------
+    t : {parse_time_types}
+        Time to use in a parse-time-compatible format
+    """
+    solar_semidiameter_rad = constants.radius / earth_distance(t)
+    return Angle(solar_semidiameter_rad.to(u.arcsec, equivalencies=u.dimensionless_angles()))
+
+
+@add_common_docstring(**_variables_for_parse_time_docstring())
+def sky_position(t='now', equinox_of_date=True):
+    """
+    Returns the apparent position of the Sun (right ascension and declination) on the
+    celestial sphere using the equatorial coordinate system, referred to the true equinox of date
+    (as default).  Corrections for nutation and aberration (for Earth motion) are included.
+
+    Parameters
+    ----------
+    t : {parse_time_types}
+        Time to use in a parse-time-compatible format
+
+    equinox_of_date : `bool`
+        If True, output is referred to the true equinox of date.  Otherwise, output is referred to
+        the J2000.0 epoch.
+    """
+    ra = apparent_rightascension(t, equinox_of_date=equinox_of_date)
+    dec = apparent_declination(t, equinox_of_date=equinox_of_date)
+    return ra, dec
+
+
+@add_common_docstring(**_variables_for_parse_time_docstring())
+def carrington_rotation_number(t='now'):
```

Comment:
Minor - can this function be considered a coordinate?  Would it be better off somewhere else?

Reply:
I suppose it is a coordinate in the sense that time is a coordinate....

Reply:
Maybe we can re-write this in Astropy time?

Reply:
I'd argue that this is a coordinate, since it's intimately related to the heliographic longitude of disk center (L0).

Reply:
ok